script.js

```diff
@@ -1,62 +1,55 @@
+const imgArr = Array(20).fill().map((_, i) => `./num${i}.png`);
+
+let clickedNumArr=[];
+let clickedIdArr=[];
+
+let randImgArr = imgArr.slice();
 
 // Randomize img locations 
 function shuffle(o) {
-  for(var j, x, i = o.length; i; j = parseInt(Math.random() * i), x = o[--i], o[i] = o[j], o[j] = x);
-  return o;
+  return o.sort(() => (Math.random() > 0.5) ? 1 : -1);
 };
-var random = shuffle(randImgArr);
-var timeOut;
 
+let random = shuffle(randImgArr);
+let timeOut;
 
-//Game play code
+// Game play code
 const images = document.querySelectorAll("#images img");
-    //flip when clicked
+
+    // Flip when clicked
     images.forEach(img => 
-      img.onclick= function () {
+      img.addEventListener('click', function (event) {
         clearTimeout(timeOut);
-        this.src=randImgArr[this.id];
+        this.src = randImgArr[this.id];
         clickedNumArr.unshift(imgArr.indexOf(randImgArr[this.id]));
         clickedIdArr.unshift(randImgArr[this.id]);
         console.log(clickedNumArr)
         console.log(clickedIdArr)
-        //check matching if two cards are clicked
-        if(clickedNumArr[1]!=null){
-          //check to delete if matching
-          if(clickedNumArr[0]==clickedNumArr[1]-10 || clickedNumArr[1]==clickedNumArr[0]-10 ){
-            
+        // Check matching if two cards are clicked
+        if (clickedNumArr[1] != null) {
+          if (clickedNumArr[0] == clickedNumArr[1]-10 || clickedNumArr[1] == clickedNumArr[0]-10) {
+            // Delete if matching
             document.getElementById(randImgArr.indexOf(clickedIdArr[0])).classList.add("delete");
             document.getElementById(randImgArr.indexOf(clickedIdArr[1])).classList.add("delete");
-            clickedNumArr.length=0;
-            clickedIdArr.length=0;
-              
-            
-          }
-          //check to reflip if unmatching  
-          else if(clickedNumArr[0]!=clickedNumArr[1]-10 || clickedNumArr[1]!=clickedNumArr[0]-10){
-            
-           setTimeout(()=>{
-           document.getElementById(randImgArr.indexOf(clickedIdArr[0])).src="./back.png";
-           document.getElementById(randImgArr.indexOf(clickedIdArr[1])).src="./back.png";
-           
-           clickedNumArr.length=0;
-           clickedIdArr.length=0;
-          },500);
-          
+            clickedNumArr.length = 0;
+            clickedIdArr.length = 0;
+          } else if (clickedNumArr[0] != clickedNumArr[1]-10 || clickedNumArr[1] != clickedNumArr[0]-10) {
+            // Reflip if unmatching
+
+            // TODO: We should disable the event listener (onclick) here so that people can't click during the timeout.
+
+            setTimeout(()=>{
+              document.getElementById(randImgArr.indexOf(clickedIdArr[0])).src="./back.png";
+              document.getElementById(randImgArr.indexOf(clickedIdArr[1])).src="./back.png";
+
+              // TODO: Then re-enable it here, so that people can click cards again.

```

Comment:
To do this, you will probably need an array of event listeners.

You can write:
```javascript
const myHandler = function (event) { /* this does some stuff here */ };

images.forEach(img => img.addEventListener('click', myHandler));
```

This will allow you to recycle the function when you want to remove the event listener and re-add it later.

Check out the documentation for [addEventListener](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) and [the MDN guide for functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Functions) for more information.